blockchain/types/tx_internal_data.go

```diff
@@ -539,20 +541,37 @@ func NewTxInternalDataWithMap(t TxType, values map[TxValueKeyType]interface{}) (
 	return nil, errUndefinedTxType
 }
 
-func IntrinsicGasPayload(gas uint64, data []byte) (uint64, error) {
+// toWordSize returns the ceiled word size required for init code payment calculation.
+func toWordSize(size uint64) uint64 {
+	if size > math.MaxUint64-31 {
+		return math.MaxUint64/32 + 1
+	}
+
+	return (size + 31) / 32
+}
+
+func IntrinsicGasPayload(gas uint64, data []byte, isContractCreation bool, rules params.Rules) (uint64, error) {
 	// Bump the required gas by the amount of transactional data
 	length := uint64(len(data))
 	if length > 0 {
 		// Make sure we don't exceed uint64 for all data combinations
 		if (math.MaxUint64-gas)/params.TxDataGas < length {
-			return 0, kerrors.ErrOutOfGas
+			return 0, ErrGasUintOverflow
+		}
+		if isContractCreation && rules.IsMantle {
```

Comment:
Can `gas + InitCodeWordGas + TxDataGas` be over `MaxUint64`? If so, it's safer to add TxDataGas before the overflow check.
```suggestion
                gas + length*params.TxDataGas
		if isContractCreation && rules.IsMantle {
```

Reply:
L558 seems the condition for L569, and L563 is for L566. Thus, I think L558 should be placed right above L569.

Reply:
@hyeonLewis Thank you! What about refactoring like this? cc. @hyunsooda 
```
func IntrinsicGasPayload(gas uint64, data []byte, isContractCreation bool, rules params.Rules) (uint64, error) {
	// Bump the required gas by the amount of transactional data
	length := uint64(len(data))
	if length == 0 {
		return gas, nil
	}
	// Make sure we don't exceed uint64 for all data combinations
	if (math.MaxUint64-gas)/params.TxDataGas < length {
		return 0, ErrGasUintOverflow
	}
	gas += length*params.TxDataGas
	
	if isContractCreation && rules.IsMantle {
		lenWords := toWordSize(length)
		if (math.MaxUint64-gas)/params.InitCodeWordGas < lenWords {
			return 0, ErrGasUintOverflow
		}
		gas += lenWords * params.InitCodeWordGas
	}
	return gas, nil
}
```

Reply:
LGTM!

Reply:
Applied at https://github.com/klaytn/klaytn/pull/1888/commits/d71af3576167c688e57ed8b6d58054446aaadd7f.